src/Buckle/Buckle/CodeAnalysis/Binding/Binder.cs

```diff
@@ -816,10 +816,10 @@ internal sealed class Binder {
             return new BoundVariableDeclarationStatement(variable, initializer);
         } else if (type.dimensions > 0 ||
             (type.isImplicit && expression.initializer is InitializerListExpressionSyntax)) {
-            var initializer = expression.initializer.kind != SyntaxKind.NullKeyword
-                ? BindInitializerListExpression(
-                    (InitializerListExpressionSyntax)expression.initializer, type)
-                : new BoundLiteralExpression(null);
+            var initializer = (expression.initializer == null ||
+                (expression.initializer is LiteralExpressionSyntax l && l.token.kind == SyntaxKind.NullKeyword))
+                ? new BoundLiteralExpression(null)
+                : BindInitializerListExpression((InitializerListExpressionSyntax)expression.initializer, type);
```

Comment:
This is a bug fix, before `int[] a = null;` would crash; unrelated to the new feature